lib/Doctrine/CouchDB/HTTP/MultipartParserAndSender.php

```diff
@@ -0,0 +1,413 @@
+                        $boundary = $info;
+
+                    } elseif (strpos($info, "error") !== false) {
+
+                        // Missing revs at the source.
+                        // Continue till the end of this document.
+                        while (strpos($this->getNextLineFromSourceConnection(), $mainBoundary) === false) ;
+
+                    } else {
+
+                        throw new \Exception("Unknown parameter with Content-Type.");
+                    }
+
+                }
+                // Doc with attachments.
+                if (strpos($value, "multipart/related") !==false) {
+
+                    if ($boundary == '') {
+                        throw new \Exception("Boundary not set for multipart/related data.");
+                    }
+
+
+                    $boundary = explode("=",$boundary,2)[1];
+
+                    try {
+                        $responses[] = $this->sendStream(
+                            'PUT',
+                            $targetPath,
+                            $mainBoundary,
+                            array('Content-Type' => 'multipart/related; boundary=' . $boundary));
+                    } catch (\Exception $e) {
+                        $responses[] = $e;
+                    }
+
+
+                } elseif ($value == 'application/json') {
+                    // JSON doc without attachment.
+                    $jsonDoc = '';
+
+                    while(trim(($jsonDoc = $this->getNextLineFromSourceConnection())) == '');
+                    array_push($docStack, trim($jsonDoc));
+
+                    // Continue till the end of this document.
+                    while (strpos($this->getNextLineFromSourceConnection(), $mainBoundary) === false) ;
+
+                } else {
+                    throw new \UnexpectedValueException("This value is not supported.");
+                }
+            } else {
+                throw new \Exception('The first line is not the Content-Type.');
+            }
+        }
+        return array($docStack, $responses);
+    }
+
+
+    /**
+     * Reads multipart data from sourceConnection and streams it to the targetConnection.
+     * Returns the body of the request or the status code in case there is no body.
+     *
+     * @param $method
+     * @param $path
+     * @param $streamEnd
+     * @param array $requestHeaders
+     * @return mixed|string
+     * @throws \Exception
+     * @throws \HTTPException
+     */
+    protected function sendStream($method, $path, $streamEnd, $requestHeaders = array())
+    {
+        $dataStream = $this->sourceConnection;
+        if ($this->targetConnection == null) {
+            $this->targetConnection = $this->targetClient->getConnection();
+        }
+
+        // Read the json doc. Use _attachments field to find the total Content-Length and create the
+        // request header with initial doc data. At present CouchDB can't handle chunked data and needs Content-Length
```

Comment:
I think this comment line spans over 80 character wide. General code standards dictates to keep comment lines under that.